cargo-dylint/tests/ci.rs

```diff
@@ -534,6 +535,63 @@ fn update() {
     });
 }
 
+#[test]
+fn verify_registered_lints() {
+    let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+    let examples_dir = manifest_dir.join("../examples");
+
+    let folders = ["general", "supplementary"];
+
+    let register_lints_re =
+        Regex::new(r"([a-zA-Z_][a-zA-Z0-9_]*)::register_lints\s*\(").expect("Invalid regex");
+
+    for folder in folders.iter() {
+        let folder_path = examples_dir.join(folder);
+
+        let lib_rs_path = folder_path.join("src").join("lib.rs");
+        let file_contents = read_to_string(&lib_rs_path)
+            .unwrap_or_else(|e| panic!("Failed to read {}: {}", lib_rs_path.display(), e));
+
+        let mut extracted_lints: Vec<String> = register_lints_re
+            .captures_iter(&file_contents)
+            .map(|cap| cap.get(1).unwrap().as_str().to_string())
+            .collect();
+        extracted_lints.sort();
+        extracted_lints.dedup();
```

Comment:
```suggestion
```
If the lints are going to be put into a `BTreeSet`, there should be no need to sort them.